### PR TITLE
mpl_sample_data

### PR DESCRIPTION
**Pull request by ocefpaf**
@mwcraig can you take a look at this recipe? (I put this together after Scholz's Beer Garten, so it may need a closer look from someone else :wink:)

### CODE REVIEW

File: recipes/mpl_sample_data/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  dev_url: https://github.com/matplotlib/matplotlib/
+  doc_url: http://matplotlib.org/contents.html
+
+extra:
+  recipe-maintainers:
+    - mdboom
+    - ocefpaf
+    - pelson
+    - tacaswell
+    - dopplershift
```
Comment (ocefpaf) [inline recipes/mpl_sample_data/meta.yaml]: I copied the `mpl` maintainers here. If anyone wants out please let me know.

File: recipes/mpl_sample_data/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  fn: matplotlib-{{ version }}.tar.gz
+  url: https://github.com/matplotlib/matplotlib/archive/v{{ version }}.tar.gz
+  sha256: aebed23921562792b68b8ca355de5abc176af4424f1987e2fa95f65e5c5e7e89
+
+build:
+  number: 0
+
+requirements:
+  run:
+    - matplotlib
```
Comment (ocefpaf) [inline recipes/mpl_sample_data/meta.yaml]: If we go the `basemap` route the data will depend on `matplotlib` and not the other way around. But if you want the data the always be present we need to change this.

**Comment by mwcraig** [comment]
@tacaswell -- not quite multiple output products form one build but would make the data available...

**Comment by mwcraig** [comment]
Looks likes windows doesn't like wildcards? https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.14965/job/th6rbrj2ys0ilh0d#L345

File: recipes/mpl_sample_data/bld.bat [hunk 3]
```diff
@@ -0,0 +1,5 @@
+set DATA_DIR="%SP_DIR%\matplotlib\mpl-data\sample_data"
+
+if not exist %DATA_DIR% mkdir %DATA_DIR%
+
+xcopy %SRC_DIR%\lib\mpl_toolkits\basemap\data\*.* %DATA_DIR% /s /e || exit 1
```
Comment (tacaswell) [inline recipes/mpl_sample_data/bld.bat]: Why is basemap in here?

**Comment by ocefpaf** [comment]
> @tacaswell -- not quite multiple output products form one build but would make the data available...

If we chose the conda feature `outputs` we could have a single recipe but we would still need this new output in all the matrix items instead of just once like in this approach.

Maybe we can tell conda that one of the recipe `outputs` is `noarch`, but I am not sure how to do that.

File: recipes/mpl_sample_data/bld.bat (hunk 3 above)
Comment (ocefpaf) [inline recipes/mpl_sample_data/bld.bat]: copy-n-pasta :flushed: fixing it now...

**Comment by msarahan** [comment]
@ocefpaf outputs in conda-build does support noarch.  I think it needs to be documented.

```
outputs:
  - name: frank
     noarch: generic
```

**Comment by ocefpaf** [comment]
> @ocefpaf outputs in conda-build does support noarch. I think it needs to be documented.

@jjhelmus just told me the same. I will try that later today before we merge this. I strongly prefer a single recipe for this!

**Comment by ocefpaf** [comment]
@msarahan please correct me if I am wrong but to use `noarch`+`outputs` efficiently we would need `conda-build 3`, right? I'm merging this to solve the current problem but I'll open an issue in both feedstocks to merge them once we are using `conda-build 3`.

**Comment by msarahan** [comment]
Cb3 allows outputs that depend on other outputs, and resolves the build
order for you. CB2.1 does not. If there's any intradependency, cb3 may be
necessary.

Cb3 may be necessary also for noarch in outputs, but that can probably be
backported.

**Pull request merged**